extension/app/scripts.babel/manifest-parser.js

```diff
@@ -0,0 +1,306 @@
+var ManifestParser = function() {
+  'use strict';
+
+  var _jsonInput = {};
+  var _manifest = {};
+  var _logs = [];
+  var _tips = [];
+  var _success = true;
+
+  var ALLOWED_DISPLAY_VALUES = ['fullscreen',
+                                 'standalone',
+                                 'minimal-ui',
+                                 'browser'];
+
+  var ALLOWED_ORIENTATION_VALUES = ['any',
+                                     'natural',
+                                     'landscape',
+                                     'portrait',
+                                     'portrait-primary',
+                                     'portrait-secondary',
+                                     'landscape-primary',
+                                     'landscape-secondary'];
+
+  function _parseString(args) {
+    var object = args.object;
+    var property = args.property;
+    if (!(property in object)) {
+      return undefined;
+    }
+
+    if (typeof object[property] !== 'string') {
+      _logs.push('ERROR: \'' + property +
+          '\' expected to be a string but is not.');
+      return undefined;
+    }
+
+    if (args.trim) {
+      return object[property].trim();
+    }
+    return object[property];
+  }
+
+  function _parseBoolean(args) {
+    var object = args.object;
+    var property = args.property;
+    var defaultValue = args.defaultValue;
+    if (!(property in object)) {
+      return defaultValue;
+    }
+
+    if (typeof object[property] !== 'boolean') {
+      _logs.push('ERROR: \'' + property +
+          '\' expected to be a boolean but is not.');
+      return defaultValue;
+    }
+
+    return object[property];
+  }
+
+  function _parseURL(args) {
+    var object = args.object;
+    var property = args.property;
+
+    var str = _parseString({object: object, property: property, trim: false});
+    if (str === undefined) {
+      return undefined;
+    }
+
+    return object[property];
```

Comment:
some of this stuff from the original bookmarklet is just crazy-town banana pants. Where does it come from originally, I wonder?